### PR TITLE
VZ-9148: Build From Source for Rancher release v2.7.2

### PR DESCRIPTION
**Pull request by Abhijithang**
This PR uptakes Build from source changes for Rancher release v2.7.2.

### CODE REVIEW

File: BUILD_FROM_SOURCE_README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+## Source Code Notes
```
Comment (mgianatagh) [inline BUILD_FROM_SOURCE_README.md:1]: We need to add a section to this document that describes the other images that may need to be rebuilt when the charts change (e.g. rancher-webhook, fleet, etc.)

File: package/Dockerfile [hunk 2]
```diff
@@ -57,16 +71,35 @@ ENV CATTLE_FLEET_MIN_VERSION=102.0.0+up0.6.0
 ENV CATTLE_RANCHER_WEBHOOK_MIN_VERSION=2.0.2+up0.3.2
 ENV CATTLE_CSP_ADAPTER_MIN_VERSION=2.0.1
 
+# Chart git repo directories
+ENV RANCHER_CHARTS_DIR=/var/lib/rancher-data/local-catalogs/v2/rancher-charts/4b40cac650031b74776e87c1a726b0484d0877c3ec137da0872547ff9b73a721
+ENV RANCHER_PARTNER_CHARTS_DIR=/var/lib/rancher-data/local-catalogs/v2/rancher-partner-charts/8f17acdce9bffd6e05a58a3798840e408c4ea71783381ecd2e9af30baad65974
+ENV RANCHER_RKE2_CHARTS_DIR=/var/lib/rancher-data/local-catalogs/v2/rancher-rke2-charts/675f1b63a0a83905972dcab2794479ed599a6f41b86cd6193d69472d0fa889c9
+
+# System chart commits
+ENV RANCHER_CHARTS_COMMIT=44e96408636e91cfbc88d3a30e5706b96021b390
```
Comment (mgianatagh) [inline package/Dockerfile]: This commit will need to be updated after the rancher-charts PR is merged

File: BUILD_FROM_SOURCE_README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+### Pinning Bundled System Charts
+
+The Rancher Dockerfile used to build the Rancher image uses `git clone` to embed Helm system charts in the image. The clone is done from a branch, but does not checkout a specific commit. There are commits to the Rancher partner chart repositories fairly regularly, which means every time we rebuild our forked Rancher image it could pull in a newer version of the system charts. This makes the Rancher image build nondeterministic with respect to the bundled system charts.
+
+To fix this problem, we have modified the [Dockerfile](package/Dockerfile#L95-L102) to checkout specific commits in the system chart repositories. When building a new version of Rancher from source, you must determine the commit ids that are used in the upstream Rancher image and set them in the [Dockerfile](package/Dockerfile#L80-L82). The easiest way to determine the commit ids is to run the upstream Rancher image interactively and use `git`
+
+For example, to find the relevant chart git commits for v2.7:
```
Comment (mgianatagh) [inline BUILD_FROM_SOURCE_README.md]: This section needs to be updated to reference v2.7.2

File: BUILD_FROM_SOURCE_README.md (hunk 1 above)
Comment (Abhijithang) [reply BUILD_FROM_SOURCE_README.md:1]: Thank you for letting me know. I missed that information. I have updated it now. Could you please check and confirm if it is fine?

File: package/Dockerfile (hunk 2 above)
Comment (Abhijithang) [reply package/Dockerfile]: Right, I merged rancher-charts PR and updated the latest commit from 2.7.2 release branch. Thank you.

File: BUILD_FROM_SOURCE_README.md (hunk 3 above)
Comment (Abhijithang) [reply BUILD_FROM_SOURCE_README.md]: Sure, updated the info. Thanks.

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**